cli/derive/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "iroha_cli_derive"
+version = "2.0.0-pre-rc.4"
+authors = ["Iroha 2 team <https://github.com/orgs/soramitsu/teams/iroha2>"]
+edition = "2021"
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[lib]
+proc-macro = true
+
+[dependencies]
+syn = {version = "1", default-features = false, features = ["full"] }
```

Comment:
`full` feature is unlikely to be needed in this simple macro

Reply:
Do you mean that as "if all of our proc macro crates use full features of syn, then syn will only be compiled once instead of many times with different feature flags"? A few of our crates had full features enabled in syn, so I followed suit. There was at least one with minimal features only though.

Reply:
@mversic Say **Full** was **A** and **B**. In crate **a** we use feature **A**, and in crate **b,** we use feature **B.** 

I was surprised to find that if you have resolver version 2, you might get better compilation times and smaller binary sizes (without LTO) if crates **a** and **b** both simultaneously depended on **Full**. 

It's the basis of `[cargo-hakari](https://crates.io/crates/cargo-hakari)`.

Reply:
Also from [here](https://rust-lang.github.io/rfcs/2957-cargo-features2.html#host-dependencies):
> Note that a dependency shared between a build-dependency and proc-macro are still unified. This is intended to help reduce build times, and is expected to be unlikely to cause problems that feature unification usually cause because they are both being built for the host platform, and are only used at build time.